tests/unit/s2n_random_test.c

```diff
@@ -794,24 +794,25 @@ static int s2n_random_rand_bytes_after_cleanup_cb(struct random_test_case *test_
 
 static int s2n_random_rand_bytes_before_init(struct random_test_case *test_case)
 {
-#if S2N_LIBCRYPTO_SUPPORTS_CUSTOM_RAND
-    /* Calling RAND_bytes will set a global random method */
-    unsigned char rndbytes[16] = { 0 };
-    EXPECT_EQUAL(RAND_bytes(rndbytes, sizeof(rndbytes)), 1);
-    const RAND_METHOD *rand_method = RAND_get_rand_method();
-    EXPECT_NOT_NULL(rand_method);
-    EXPECT_NOT_EQUAL(rand_method->bytes, s2n_openssl_compat_rand);
-
-    EXPECT_SUCCESS(s2n_init());
+#if S2N_LIBCRYPTO_SUPPORTS_ENGINE
+    if (!s2n_libcrypto_is_boringssl() && !s2n_libcrypto_is_libressl() && !s2n_libcrypto_is_awslc() && !s2n_libcrypto_is_fips()) {
```

Comment:
should we make this a callable function so we're not duplicating it in tests? I see this copied 3 times in the PR.

Reply:
Yep probably worth doing. Since older versions of openssl is when we care to setup our custom random I think we want `is_openssl()`.

Is my assumption true and do we have a `is_openssl()` function we can rely on?

Reply:
no i'm more suggesting we have a `bool s2n_libcrypto_requires_custom_rand(void)` function that consolidates this logic in a single place.

Reply:
Sure we can add:

```
bool s2n_libcrypto_requires_custom_rand(void) {
    !s2n_libcrypto_is_boringssl() &&
    !s2n_libcrypto_is_libressl() &&
    !s2n_libcrypto_is_awslc() &&
    !s2n_libcrypto_is_fips()
}
```
But I am wondering if there is a reason why it cant just be the following:

```
bool s2n_libcrypto_requires_custom_rand(void) {
    is_openssl() &&
    !s2n_libcrypto_is_fips()
}
```

Reply:
As long as those are equivalent values I think that's fine

Reply:
@torben-hansen are you able to provide context on this?

Are the two declarations above equivalent and is it possible to detect openssl libcrypto?

Reply:
I'm not sure there is a direct "this is the real/true OpenSSL and not one of its forks". Such a function would probably try to exclude the known forks using the same sequence sequence of `is_<openssl fork>()`.

I would exclude known forks myself.